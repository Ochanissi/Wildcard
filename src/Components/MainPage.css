@font-face {
    font-display: swap;
    font-family: "Times New Roman", Times, serif;
}

h1 {
    font-weight: 900;
    color: #fff;
}

