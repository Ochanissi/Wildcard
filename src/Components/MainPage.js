import React, { Component } from 'react';

import CardList from '../Components/CardList';
import SearchBox from '../Components/SearchBox'
import Scroll from '../Components/Scroll';
import ErrorBoundry from '../Components/ErrorBoundry';
import Header from '../Components/Header';
import './MainPage.css';

class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestCharacters();
    }

    filterCharacters = () => {
        return this.props.characters.filter(character => {
            return character.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        })
    }

    render() {
        const { onSearchChange, isPending } = this.props;

        if (isPending) {
            return <h1>Loading</h1>
        } else {
            return (
                <div className='tc'>
                    <Header />
                    <SearchBox searchChange={onSearchChange}/>
                    <Scroll>
                        { isPending ? <h1>Loading</h1> :
                        <ErrorBoundry>
                            <CardList characters={this.filterCharacters()}/>
                        </ErrorBoundry>
                        }
                    </Scroll>
                </div>
            );
        }
    }
}

export default MainPage;