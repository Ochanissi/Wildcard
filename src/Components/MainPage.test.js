import { shallow, mount, render } from 'enzyme';
import React from 'react';
import MainPage  from '../Components/MainPage';
import { jsxEmptyExpression, exportAllDeclaration } from '@babel/types';


let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestCharacters: jest.fn(),
        characters: [],
        searchField: '',
        isPending: false
    }

    wrapper = shallow(<MainPage { ...mockProps } />);
})

it('renders MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot();
})

it('renders characters correctly', () => {
    const mockProps2 = {
        onRequestCharacters: jest.fn(),
        characters: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'John',
        isPending: false
    }
    wrapper2 = shallow(<MainPage { ...mockProps } />);
    expect(wrapper2.instance().filterCharacters([])).toEqual([]);
    expect(wrapper2.instance().filterCharacters([])).toEqual([]);
})