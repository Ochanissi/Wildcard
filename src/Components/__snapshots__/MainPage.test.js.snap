// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `ShallowWrapper {}`;
